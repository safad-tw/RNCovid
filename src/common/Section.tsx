import * as React from 'react';
import {
  Text,
  View,
  StyleSheet,
  useColorScheme
} from 'react-native';
import {
  Colors
} from 'react-native/Libraries/NewAppScreen';

const Section: React.FC<{
  title: string;
}> = ({ children, title }) => {
  const isDarkMode = useColorScheme() === 'dark';
  return (
    <View style={styles.sectionContainer}>
      <Text
        style={[
          styles.sectionTitle,
          {
            color: isDarkMode ? Colors.white : Colors.black,
          },
        ]}>
        {title}
      </Text>
      <Text
        style={[
          styles.sectionDescription,
          {
            color: isDarkMode ? Colors.light : Colors.dark,
          },
        ]}>
        {children}
      </Text>
    </View>
  );
};


const styles = StyleSheet.create({
  sectionContainer: {
    marginTop: 32,
    paddingHorizontal: 10,
    width: '100%'
  },
  headerContainer: {
    height: 60
  },
  headerTitle: {
    fontSize: 22,
    textAlign: 'center',
    fontFamily: "Avenir-heavy",
    margin: 5
  },

  sectionTitle: {
    fontSize: 18,
    textAlign: 'center',
    fontFamily: "Avenir-black",
  },
  sectionDescription: {
    marginTop: 8,
    fontSize: 18,
    fontWeight: '400',
  },
  highlight: {
    fontWeight: '700',
  },
});

export default Section;