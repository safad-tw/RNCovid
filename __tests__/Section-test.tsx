import * as React from "react";
import renderer from "react-test-renderer";
import { render, fireEvent } from '@testing-library/react-native';
import '@testing-library/jest-dom'
import Section from "../src/common/Section";

test("Render component", () => {
    const tree = renderer.create(<Section title={'Global'}/>).toJSON();
    expect(tree).not.toBeNull()
  });

  test('Renders snapshot as expected', () => {
    const tree = renderer.create(<Section title={'Global'}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });