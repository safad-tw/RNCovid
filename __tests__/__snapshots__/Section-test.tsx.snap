// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders snapshot as expected 1`] = `
<View
  style={
    Object {
      "marginTop": 32,
      "paddingHorizontal": 10,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Avenir-black",
          "fontSize": 18,
          "textAlign": "center",
        },
        Object {
          "color": "#000",
        },
      ]
    }
  >
    Global
  </Text>
  <Text
    style={
      Array [
        Object {
          "fontSize": 18,
          "fontWeight": "400",
          "marginTop": 8,
        },
        Object {
          "color": "#444",
        },
      ]
    }
  />
</View>
`;
